model/host/host.go

```diff
@@ -2461,3 +2461,167 @@ func (h *Host) IsSubjectToHostCreationThrottle() bool {
 
 	return true
 }
+
+func GetPaginatedRunningHosts(hostID, distroID, currentTaskID string, statuses []string, startedBy string, sortBy string, sortDir, page, limit int) ([]Host, *int, int, error) {
+	// PIPELINE FOR ALL RUNNING HOSTS
+	runningHostsPipeline := []bson.M{
+		{
+			"$match": bson.M{StatusKey: bson.M{"$ne": evergreen.HostTerminated}},
+		},
+		{
+			"$lookup": bson.M{
+				"from":         task.Collection,
+				"localField":   RunningTaskKey,
+				"foreignField": task.IdKey,
+				"as":           "task_full",
+			},
+		},
+		{
+			"$unwind": bson.M{
+				"path":                       "$task_full",
+				"preserveNullAndEmptyArrays": true,
+			},
+		},
+	}
+
+	// CONTEXT
+	env := evergreen.GetEnvironment()
+	ctx, cancel := env.Context()
+	defer cancel()
+
+	// TOTAL RUNNING HOSTS COUNT
+	countPipeline := []bson.M{}
+	for _, stage := range runningHostsPipeline {
+		countPipeline = append(countPipeline, stage)
```

Comment:
why do we add a count after every stage in running hosts pipeline? I think a running host is any host that is running a task so we can probably just count how many hosts have a `task_full` key that is not empty

Reply:
I copied @john-m-liu's logic from [GetTasksByVersion](https://github.com/evergreen-ci/evergreen/blob/master/model/task/task.go#L2374) for counting results from a pipeline. 

I do not believe a running host = any host with a running task. if you go to the hosts page in evergreen, you can see that roughly half of the hosts displayed do not have a running task. all hosts shown on this page are considered "running". The original query for running hosts can be seen [here](https://github.com/evergreen-ci/evergreen/blob/master/model/host/host.go#L1771) and it does not specify that a host must have a running task, although it can have a running task.